interop/e2e_tests/postie_test.go

```diff
@@ -130,3 +130,165 @@ func TestPostieInboxRelay(t *testing.T) {
+	require.NoError(t, err)
+	require.True(t, msgPassIter.Next())
+
+	require.NoError(t, wait.For(context.Background(), time.Second/2, func() (bool, error) {
+		return testSuite.PostieA.OutboxStorageRoot(testSuite.ChainIdB) != oldStorageRoot, nil
+	}))
+
+	// Relay this message onto chain A
+	inbox, err := bindings.NewCrossL2Inbox(predeploys.CrossL2InboxAddr, testSuite.OpSysA.Clients["sequencer"])
+	require.NoError(t, err)
+
+	outboxRoot := testSuite.PostieA.OutboxStorageRoot(testSuite.ChainIdB)
+	t.Logf("outbox root: %s", outboxRoot)
+
+	senderOpts, err = bind.NewKeyedTransactorWithChainID(sender, big.NewInt(int64(testSuite.ChainIdA)))
+	// Purposefully setting the gasLimit low to induce an out of gas revert
+	senderOpts.GasLimit = 1
+	require.NoError(t, err)
+	_, err = inbox.RunCrossL2Message(senderOpts,
+		bindings.TypesSuperchainMessage{
+			Nonce:       big.NewInt(0), // first message
+			SourceChain: common.BigToHash(big.NewInt(int64(testSuite.ChainIdB))),
+			TargetChain: common.BigToHash(big.NewInt(int64(testSuite.ChainIdA))),
+			From:        senderAddr,
+			To:          senderAddr,
+			GasLimit:    big.NewInt(25_000),
+			Data:        []byte{},
+			Value:       big.NewInt(params.Ether / 2),
+		},
+		outboxRoot,
+		genMPTProof(t, outboxRoot, msgPassIter.Event, testSuite.OpSysB.Clients["sequencer"]),
+	)
+	require.Error(t, err)
+
+	// Resetting gasLimit to a sufficient amount to assert replayability
+	senderOpts, err = bind.NewKeyedTransactorWithChainID(sender, big.NewInt(int64(testSuite.ChainIdA)))
+	require.NoError(t, err)
+	tx, err = inbox.RunCrossL2Message(senderOpts,
+		bindings.TypesSuperchainMessage{
+			Nonce:       big.NewInt(0), // first message
+			SourceChain: common.BigToHash(big.NewInt(int64(testSuite.ChainIdB))),
+			TargetChain: common.BigToHash(big.NewInt(int64(testSuite.ChainIdA))),
+			From:        senderAddr,
+			To:          senderAddr,
+			GasLimit:    big.NewInt(25_000),
+			Data:        []byte{},
+			Value:       big.NewInt(params.Ether / 2),
+		},
+		outboxRoot,
+		genMPTProof(t, outboxRoot, msgPassIter.Event, testSuite.OpSysB.Clients["sequencer"]),
+	)
+	require.NoError(t, err)
+
+	_, err = wait.ForReceiptOK(context.Background(), testSuite.OpSysA.Clients["sequencer"], tx.Hash())
+	require.NoError(t, err)
+}
+
+// Asserting that a message cannot be relayed on a chain if the _msg.targetChain does
+// not match block.chainid
+func TestPostieInboxWrongChainId(t *testing.T) {
+	testSuite := createE2ETestSuite(t)
+
+	// wait for the first storage root of chain B to change
+	var oldStorageRoot common.Hash
+	require.NoError(t, wait.For(context.Background(), time.Second/2, func() (bool, error) {
+		oldStorageRoot = testSuite.PostieA.OutboxStorageRoot(testSuite.ChainIdB)
+		return oldStorageRoot != common.Hash{}, nil
+	}))
+
+	outbox, err := bindings.NewCrossL2Outbox(predeploys.CrossL2OutboxAddr, testSuite.OpSysB.Clients["sequencer"])
+	require.NoError(t, err)
+
+	// Transfer 0.5 ETH from Bob's account from Chain B -> A
```

Comment:
nit: this comment should be updated that this message is being sent to chain id 42 and not chain A